### PR TITLE
Dark mode: accordions

### PR DESCRIPTION
**Pull request by louismaximepiton**
_Note: Please transform `- [ ]` into `- (NA)` in the description when things are not applicable_

### Related issues

#2223.

### Description

Adding the accordion as dark variant

I didn't mention any migration note or w/ever, maybe we should think about a section or something to put everything related to dark mode ?

### Motivation & Context

Moving forward on dark theme components.

### Types of change

- New feature (non-breaking change which adds functionality)

### Live previews

- <https://deploy-preview-2259--boosted.netlify.app/docs/5.3/components/accordion>

### CODE REVIEW

File: site/content/docs/5.3/components/accordion.md [hunk 1]
```diff
@@ -202,6 +202,14 @@ Omit the `data-bs-parent` attribute on each `.accordion-collapse` to make accord
 
 ## Dark variant
 
+{{< deprecated-in "5.3.3" >}}
+
+{{< callout warning >}}
+New dark accordions in v5.3.3 — We’ve deprecated `.accordion-dark` in favor of the new `data-bs-theme="dark"`. Add `data-bs-theme="dark"` to the `.accordion` or any ancestor element to enable a component-specific color mode. [Learn more about our color modes]({{< docsref "/customize/color-modes" >}}).
+
+`.accordion-dark` will be removed in v6.
```
Comment (louismaximepiton) [inline site/content/docs/5.3/components/accordion.md:210]: Not quite sure about this sentence, feel free to remove.

**Comment by louismaximepiton** [comment]
Dark variant seem to be as before, however, the text color for the code sample isn't the right one. It's because we removed the `*-dark` rule. We can have a pretty similar look as bafore if we had `data-bs-theme="dark"` to the example but I tried to avoid it in this PR. Maybe it's not feasible anymore and we should keep to what Bootstrap done and try to enhance it.

Maybe we should drop the dark variants as Bs and maintain only the local dark mode.

**Comment by MewenLeHo** [comment]
screenshot-deploy-preview-2259--boosted netlify app-2023 09 25-11_17_02

This section of the page is a bit confusing to me 🤔 

First, having both tags `Added in x.x.x` and `Deprecated in x.x.x` may puzzled some of our users. Should we keep only one tag, i.e. the most recent one?

Then it's written that `.accordion-dark` has been deprecated, will be removed in v6 and finally there is a paragraph asking you to add this class to your code. There is even a code example using this deprecated class.

It sounds understandable when you know the subject and the way we work on color modes but from a simple user perspective it may look a bit weird and confusing. Especially for peaople with attention disorder for instance.

**Comment by MewenLeHo** [comment]
screenshot-deploy-preview-2259--boosted netlify app-2023 09 25-12_01_18
`#666` on `#000` will cause a contrast issue.

**Comment by julien-deramond** [comment]
I tried something in https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2259/commits/ed5bf7514d42b49bd3d5ae7635d0cc4aac69f039.

## Dark variant retro compatibility

I've reintroduced our root rules for dark variants in `_root.scss`:
```scss
:not([data-bs-theme="dark"]) {
  [class*="bg-black"],
  [class*="-dark"]:not(.border-dark):not(.text-dark):not(.btn-dark):not(.focus-ring-dark):not(.link-underline-dark):not(.link-dark),
  [class*="bg-secondary"] {
  }
}
```
but it's now scoped with a `:not([data-bs-theme="dark"])` so that only dark variants execute them.

It should allow the following mapping:
- In light mode, accordions in general in https://deploy-preview-2259--boosted.netlify.app/docs/5.3/components/accordion/ should have the same rendering as https://boosted.netlify.app/docs/5.3/components/accordion/
- In light mode, dark accordions in https://deploy-preview-2259--boosted.netlify.app/docs/5.3/components/accordion/#dark-variant should have the same rendering as https://boosted.netlify.app/docs/5.3/components/accordion/#dark-variant
- In dark mode, accordions in general in https://deploy-preview-2259--boosted.netlify.app/docs/5.3/components/accordion should have the new dark mode rendering
- In dark mode, dark accordions in https://deploy-preview-2259--boosted.netlify.app/docs/5.3/components/accordion/#dark-variant should have the same rendering as in light mode (dark accordions) and should be on black background (not `#141414`).

IDK TBH if it's going to be feasible for all our components but maybe but can try it.

## Dark variant deprecation

- As pointed out by @MewenLeHo, having "deprecated" and "added" is difficult to understand so I've dropped out the "added" mention
- I've changed a little bit the wording to be more generic. Note that this is not yet a reusable message but could be with the right class as a parameter if this generic message is enough for a good understanding.

**Pull request merged**